tasks/main.yml

```diff
@@ -6,31 +6,57 @@
     - "{{ ansible_os_family }}.yml"
   tags: install
 
-- name: Generate configuration file
+- name: Test directories presence
+  stat:
+    path: "{{ item }}"
+  register: _dir
+  with_items:
+    - "/etc/gridinit.d/{{ openio_kibana_namespace }}"
+    - "{{ openio_kibana_sysconfig_dir }}"
+    - "{{ openio_kibana_volume }}"
+    - "{{ openio_kibana_pid_dir }}"
+    - "{{ openio_kibana_log_dir }}"
+
+- name: Ensure directories exists
+  file:
+    path: "{{ item.item }}"
+    state: directory
+    owner: openio
+    group: openio
+    mode: 0755
+    recurse: true
+  with_items: "{{ _dir.results }}"
+  when: not item.stat.exists
+  tags: install
+
+- name: Generate configuration files
   template:
-    src: "kibana.yml.j2"
-    dest: "{{ openio_kibana_confdir }}/kibana.yml"
+    src: "{{ item.src }}"
+    dest: "{{ item.dest }}"
     owner: openio
     group: openio
     mode: 0644
+  with_items:
+    - src: "kibana.yml.j2"
+      dest: "{{ openio_kibana_sysconfig_dir }}/kibana.yml"
+    - src: "gridinit_kibana.conf.j2"
+      dest: "{{ openio_kibana_gridinit_dir }}/{{ openio_kibana_gridinit_file_prefix}}{{ openio_kibana_servicename }}.conf"
   register: _kibana_conf
-  tags: configure
 
 - name: Restart kibana
-  service:
-    name: kibana
-    state: "{{ 'restarted' if openio_kibana_provision_only else 'started' }}"
-    enabled: "{{ openio_kibana_service_enabled }}"
+  shell: |
+    gridinit_cmd reload
+    gridinit_cmd restart {{ openio_kibana_namespace }}-{{ openio_kibana_servicename }}
   tags: configure
   when:
     - _kibana_conf is changed
+    - not openio_kibana_provision_only
 
-- name: Ensure kibana is listening
+- name: Wait for kibana to be available
   wait_for:
     host: "{{ openio_kibana_bind_address }}"
     port: "{{ openio_kibana_bind_port }}"
     delay: 10
-  tags: configure
```

Comment:
tasks must be tagged install or configure
rpm and mkdir are `install`
anything else is configure

Reply:
fixed